kyo-core/shared/src/main/scala/kyo/core.scala

```diff
@@ -106,6 +106,23 @@ object core:
         end handle
     end extension
 
+    inline def eval[T, S, U >: T, S2](v: => T < S)(
+        inline resume: (Safepoint[S & S2], () => U < (S & S2)) => U < (S & S2),
+        inline done: (Safepoint[S & S2], T) => U < (S & S2) = (s: Safepoint[S & S2], v: T) => v,
+        inline suspend: U < (S & S2) => U < (S & S2) = (v: U < (S & S2)) => v
```

Comment:
I'm sorry for this symbol salad 😂 I tried defining an `EvalHandler` with a more readable API but I couldn't avoid overhead with it.